.pre-commit-config.yaml

```diff
@@ -0,0 +1,35 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+default_language_version:
+    python: python3.7
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.4.0
+    hooks:
+    -   id: no-commit-to-branch
+    -   id: check-executables-have-shebangs
+    -   id: check-ast
+    -   id: check-merge-conflict
+    -   id: check-toml
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+    -   id: check-case-conflict
+    -   id: check-json
+    -   id: check-merge-conflict
+    -   id: check-symlinks
+    -   id: pretty-format-json
+        args:
+        - --autofix
+-   repo: https://gitlab.com/pycqa/flake8
+    rev: 3.8.4
+    hooks:
+    -   id: flake8
+-   repo: https://github.com/Lucas-C/pre-commit-hooks
+    rev: v1.1.7
+    hooks:
+    -   id: forbid-crlf
```

Comment:
How does this work on Windows where git will auto convert crlf to lf before committing? Would it cause a error?

Reply:
None of my mates at work that are on windows have complained but I'll fire up a VM and doublecheck this personally. That'll probably happen tomorrow though (or maybe much later tonight is I can't sleep or something).

I'm pretty sure the lf<->crlf conversion that git does on windows is done in such way that the hook only sees the actual content going to repo (which would have been converted back to lf)

Reply:
hmm, this is getting way more complicated than just installing git and python for windows, what kind of env are the windows people actually running for development ?? If they're on WSL or cygwin they probably do not have the CRLF conversion even enabled if not how do they even enable the pre-commit hooks (ie where do they get `make` from, for example...)

Anyway gits own documentation says the conversion is done before committing and the hook sees the to-be-committed content so this *shoud* be transparent.

Reply:
Personally I use msys and I set `autocrlf=input`.